### ISSUE TITLE
Accessibility: Error message on My Account edit address page not announced to screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43635]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the My Account page, if there is an error submitting the form, the page refreshes rather than submitting the form with ajax, and there is a visual error message but screen reader users are loaded at the top of the page and do not hear the message.
<img width="854" alt="Screenshot of form error message" src="https://github.com/woocommerce/woocommerce/assets/13153456/a1e606e3-1fc6-45c3-a660-ed2569017796">

### Expected behavior

When a form error is present, there should be an auditory message for screen reader users to know that there was an error.

You can either do this by:
A) When the page reloads, set keyboard focus on the error message so it is read out. OR
B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it after the page loads.

You may also consider changing this form to submit with ajax so screen reader users are not sent to the top of the page as happens with a refresh.

### Actual behavior

Screen reader users don't easily know when a an error occurs.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to the My Account > Addresses page.
3. Click edit on one of the addresses.
4. Leave at least one required field blank and submit the form.
5. Listen as the page reloads and observe there's no announcement about the error.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by amberhinds** [comment]
This issue also applies to the error message on the Account Details page, which is likely using the same component.
<img width="1101" alt="Screenshot of error message on account details page" src="https://github.com/woocommerce/woocommerce/assets/13153456/28f55565-3aa3-4e33-89b2-1e3e3310e046">
